### PR TITLE
Enabling Hal Plugin

### PR DESCRIPTION
**Pull request by matteoscurati**
Fixes # .

Changes proposed in this pull request:
- integrate HAL Plugin on the Snapshot UI

## Goal
To integrate HAL on the Snapshot UI, so that users can set up a trigger to notify them when a proposal is created or ends for a any given space and receive notifications over email, Telegram, Discord, Slack, or connect it to webhooks to create further automation.

## Flow
1. User on the Snapshot website wants to be notified when a proposal is created or ends for a specific space.
Screenshot%202021-12-16%20at%2017 13 11
2. User gets redirect to a custom made recipe on the HAL website, e.g. https://9000.hal.xyz/recipes/snapshot-follow-new-proposals?space-id=test3.shot.eth&status=end.
3. All fields in the recipe are already pre-filled and a custom template is ready to be used right away. The users choose the relevant action (Email, Telegram etc).
Screenshot%202021-12-16%20at%2017 15 13
4. From HAL website, the user can see their triggers and interact with them (edit, delete etc).

### CODE REVIEW

File: src/components/Plugin/Hal/CustomBlock.vue [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<script setup>
+const props = defineProps({
+  space: Object
+});
+
+const getSpaceId = () => {
```
Comment (samuveth) [inline src/components/Plugin/Hal/CustomBlock.vue:6]: You should be using `computed` here like this.
```
const spaceId = computed(() => props.space.id)
```
But in this case you can just use `props.space.id` inside script and `space.id` inside template.

File: src/components/Plugin/Hal/CustomBlock.vue [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  const spaceId = props.space.id;
+  return spaceId;
+};
+
+const getSpaceName = () => {
+  const spaceName = props.space.name;
+  return spaceName;
+};
+
+const getHalUrl = () => {
```
Comment (samuveth) [inline src/components/Plugin/Hal/CustomBlock.vue:16]: Also here use computed
```
const halUrl = computed(() => `https://9000.staging.hal.xyz/recipes/snapshot-follow-new-proposals?space-id=${props.space.id}&status=end`)

File: src/components/Plugin/Hal/CustomBlock.vue [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  const spaceId = getSpaceId();
+  return `https://9000.staging.hal.xyz/recipes/snapshot-follow-new-proposals?space-id=${spaceId}&status=end`;
+};
+
+const getLogoUrl = () => {
```
Comment (samuveth) [inline src/components/Plugin/Hal/CustomBlock.vue:21]:
```
const logoUrl = "https://hal-snapshot-plugin.s3.eu-west-1.amazonaws.com/hal_bell.svg"
```

File: src/components/Plugin/Hal/CustomBlock.vue [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+const getSpaceId = () => {
+  const spaceId = props.space.id;
+  return spaceId;
+};
+
+const getSpaceName = () => {
+  const spaceName = props.space.name;
+  return spaceName;
+};
```
Comment (samuveth) [inline src/components/Plugin/Hal/CustomBlock.vue:14]: Also here I would just use `props.space.name` inside script and `space.name` inside template.

File: src/components/Plugin/Hal/CustomBlock.vue [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+const getHalUrl = () => {
+  const spaceId = getSpaceId();
+  return `https://9000.staging.hal.xyz/recipes/snapshot-follow-new-proposals?space-id=${spaceId}&status=end`;
+};
```
Comment (samuveth) [inline src/components/Plugin/Hal/CustomBlock.vue:19]:
```suggestion
const halUrl = computed(() => `https://9000.staging.hal.xyz/recipes/snapshot-follow-new-proposals?space-id=${props.space.id}&status=end`)
```

**Review by bonustrack — APPROVED** [review]
Looks good!

**Pull request merged**